pkg/dnshelpers/util.go

```diff
@@ -14,6 +16,89 @@ import (
 	"k8s.io/klog"
 )
 
+// GetEscapedPreferredInternalIPAddressForNodeName returns the first internal ip address of the correct family with escaping
+// for ipv6.
+func GetEscapedPreferredInternalIPAddressForNodeName(network *configv1.Network, node *corev1.Node) (string, error) {
+	address, family, err := GetPreferredInternalIPAddressForNodeName(network, node)
+	if err != nil {
+		return "", err
+	}
+	if family == "tcp6" {
+		return "[" + address + "]", nil
+	} else {
+		return address, nil
+	}
+}
+
+// GetPreferredInternalIPAddressForNodeName returns the first internal ip address of the correct family and the family
+func GetPreferredInternalIPAddressForNodeName(network *configv1.Network, node *corev1.Node) (string, string, error) {
+	ipFamily, err := GetPreferredIPFamily(network)
+	if err != nil {
+		return "", "", err
+	}
+
+	for _, currAddress := range node.Status.Addresses {
+		if currAddress.Type == corev1.NodeInternalIP {
+			switch ipFamily {
+			case "tcp4":
+				isIPv4, err := IsIPv4(currAddress.Address)
+				if err != nil {
+					return "", "", err
+				}
+				if isIPv4 {
+					return currAddress.Address, ipFamily, nil
```

Comment:
if an IPv4 address in IPv6 format, `isIPv4` will be true, but you will need the brackets.
`net.IP`'s `String()`  will return a canonical IP address string that will match the ipFamily type.

Reply:
node.Status.Addresses should never have IPv4 addresses in IPv6 format (other things will almost definitely break if it does)